src/pages/LoginForms/ControlledComponent.jsx

```diff
@@ -0,0 +1,115 @@
+import React, { useState } from "react";
+import { useReducer } from "react";
+
+export default function ControlledComponent() {
+  const [state, dispatch] = useReducer(reducer, {
+    login: "",
+    password: "",
+    validLogin: true,
+    validPassword: true,
+  });
+
+  function reducer(state, action) {
+    switch (action.type) {
+      case "setLogin":
+        return {
+          ...state,
+          login: action.payload.login,
+        };
+      case "setPassword": {
+        return {
+          ...state,
+          password: action.payload.password,
+        };
+      }
+      case "setIsValidLogin": {
+        return {
+          ...state,
+          validLogin: action.payload.validLogin,
+        };
+      }
+      case "setIsValidPassword": {
+        return {
+          ...state,
+          validPassword: action.payload.validPassword,
+        };
+      }
+      default: {
+        return state;
+      }
+    }
+  }
+
+  const handleChangeLogin = (event) => {
+    const login = event.target.value;
+    dispatch({ type: "setLogin", payload: { login: login } });
+    dispatch({ type: "setIsValidLogin", payload: { validLogin: true } });
+
+    if (!login.length || login.length > 15) {
+      dispatch({ type: "setIsValidLogin", payload: { validLogin: false } });
+      return;
```

Comment:
This return statement is redundant. Please remove it.

Reply:
fixed